monai/apps/pathology/metrics.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2020 - 2021 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from typing import Dict, List, Tuple, Union
+
+import numpy as np
+
+from monai.apps.pathology.utils import PathologyProbNMS
+from monai.data.image_reader import WSIReader
+from monai.metrics import compute_fp_tp_probs, compute_froc_curve_data, compute_froc_score
+from monai.utils import optional_import
+
+measure, _ = optional_import("skimage.measure")
+ndimage, _ = optional_import("scipy.ndimage")
+
+
+class LesionFROC:
+    """
+    Evaluate with Free Response Operating Characteristic (FROC) score.
+
+    Args:
+        data: either the list of dictionaries containg probability maps (inference result) and
+            tumor mask (ground truth), as below, or the path to a json file containing such list.
+            `{
+            "prob_map": "path/to/prob_map_1.npy",
+            "tumor_mask": "path/to/ground_truth_1.tiff",
+            "level": 6,
+            "pixel_spacing": 0.243
+            }`
+        grow_distance: Euclidean distance (in micrometer) by which to grow the label the ground truth's tumors.
+            Defaults to 75, which is the equivalent size of 5 tumor cells.
+        itc_diameter: the maximum diameter of a region (in micrometer) to be considered as an isolated tumor cell.
+            Defaults to 200.
+        eval_thresholds: the false positive rates for calculating the average sensitivity.
+            Defaults to (0.25, 0.5, 1, 2, 4, 8) which is the same as the CAMELYON 16 Challenge.
+        image_reader_name: the name of library to be used for loading whole slide imaging, either CuCIM or OpenSlide.
+            Defaults to CuCIM.
+
+    """
+
+    def __init__(
+        self,
+        data: Union[List[Dict], str],
+        grow_distance: int = 75,
+        itc_diameter: int = 200,
+        eval_thresholds: Tuple = (0.25, 0.5, 1, 2, 4, 8),
+        image_reader_name: str = "cuCIM",
+    ) -> None:
+
+        if isinstance(data, str):
+            self.data = self._load_data(data)
+        else:
+            self.data = data
+        self.grow_distance = grow_distance
+        self.itc_diameter = itc_diameter
+        self.eval_thresholds = eval_thresholds
+        self.image_reader = WSIReader(image_reader_name)
+        self.nms = PathologyProbNMS(
+            sigma=0.0,
+            prob_threshold=0.5,
+            box_size=48,
+        )
+
+    def _load_data(self, file_path: str) -> List[Dict]:
```

Comment:
could you remove `_load_data`, this data list should be read before this metric computation, to be consistent with the other metric modules

Reply:
this is to be consistent across the project. the data working group is proposing prototypes for data representation, before we have a concrete design we shouldn't assume a JSON encoded file here